### PR TITLE
ci: build binary

### PR DESCRIPTION
**Pull request by chesedo**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,6 +98,29 @@ commands:
...
+    steps:
+      - run:
+          name: Make artifact
+          command: |
+            mkdir shuttle
+            mv target/<< parameters.target >>/release/cargo-shuttle<< parameters.suffix >> shuttle/shuttle<< parameters.suffix >>
+            mv LICENSE shuttle/
+            mv README.md shuttle/
+            mkdir artifacts
+            tar -cvzf artifacts/shuttle-${CIRCLE_TAG}-<< parameters.target >>.tar.gz shuttle
```
Comment (tsar-boomba) [inline .circleci/config.yml:119]:
```suggestion
            tar -cvzf artifacts/cargo-shuttle-${CIRCLE_TAG}-<< parameters.target >>.tar.gz shuttle
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -97,6 +98,29 @@ commands:
...
+      - persist_to_workspace:
+          root: artifacts
+          paths:
+            - shuttle-${CIRCLE_TAG}-<< parameters.target >>.tar.gz
```
Comment (tsar-boomba) [inline .circleci/config.yml:123]: For better support from tools like cargo-binstall, use {crate_name}-{version}-{target}.tar.gz
```suggestion
            - cargo-shuttle-${CIRCLE_TAG}-<< parameters.target >>.tar.gz
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -97,6 +98,29 @@ commands:
...
+      suffix:
+        description: "Suffix that is on the binary"
+        type: string
+        default: ""
+    steps:
+      - run:
+          name: Make artifact
+          command: |
+            mkdir shuttle
+            mv target/<< parameters.target >>/release/cargo-shuttle<< parameters.suffix >> shuttle/shuttle<< parameters.suffix >>
```
Comment (tsar-boomba) [inline .circleci/config.yml:115]: This also needed to be changed for install support
```suggestion
            mv target/<< parameters.target >>/release/cargo-shuttle<< parameters.suffix >> shuttle/cargo-shuttle<< parameters.suffix >>
```

Comment (chesedo) [reply .circleci/config.yml:115]: Haha, thanks a million for this @tsar-boomba

**Comment by brokad** [comment]
Artifacts work for me on macOS, bar the fact they're not signed and thus require the standard override hoops to get working.

**Pull request merged**